File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_VASQUEZ_GABRIEL/Practica 3 Aux/GENERECIDAD/Ejercicio1/Caja.java
package Genericidad1;

public class Caja<T> {
    private T contenido;

    public Caja() {
    }

    public Caja(T c) {
        contenido = c;
    }

    public T obtener() {
        return contenido;
    }

    public void guardar(T contenido) {
        this.contenido = contenido;
    }

    public void mostrar() {
        System.out.println("El contenido es: " + contenido);
    }
}

File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_VASQUEZ_GABRIEL/Practica 3 Aux/GENERECIDAD/Ejercicio1/Caja.py
from typing import TypeVar, Generic

T = TypeVar('T')

class Caja(Generic[T]):
    def __init__(self, contenido: T = None):
        self.contenido = contenido

    def obtener(self) -> T:
        return self.contenido

    def guardar(self, contenido: T) -> None:
        self.contenido = contenido

    def mostrar(self) -> None:
        print("El contenido es:", self.contenido)
if __name__ == "__main__":
    C1 = Caja[str]("Carta de cumpleaños")
    C1.guardar("Carta de Recomendación")

    C2 = Caja (250)
    C2.guardar(250)

    print("--------")
    print(C1.obtener())
    C1.mostrar()

    print("--------")
    print(C2.obtener())
    C2.mostrar()
